packages/mui-material/src/StepContent/StepContent.js

```diff
@@ -110,9 +110,7 @@ const StepContent = React.forwardRef(function StepContent(inProps, ref) {
       ownerState={ownerState}
       {...other}
     >
-      <TransitionSlot as={TransitionComponent} {...transitionProps}>
```

Comment:
Lets revert this change, `TransitionComponent` should still be passed through `as`